### ISSUE TITLE
Option '--trace' doesn't work with module 'unittest'

### ISSUE THREAD

**Issue by guocb** [pytest-dev/pytest#3823]
Thanks first for adding option '--trace'. That's very helpful to me.

I tried it and it does work with _test functions_. However, it doesn't work with tests based on python unittest module, e.g.

```Python
import unittest

class TestFoo(unittest.TestCase):
    def test_1(self):
        assert True
```

The output is:
```
bash-4.4$ pytest --trace tmp/test1.py
============================================================ test session starts ============================================================
platform linux -- Python 3.6.5, pytest-3.7.1, py-1.5.3, pluggy-0.7.1
rootdir: /data/data/com.termux/files/home, inifile:
plugins: interactive-0.1.4
collected 1 item

tmp/test1.py .                                                                                                                        [100%]

========================================================= 1 passed in 0.01 seconds ==========================================================
```

**Comment by guocb** [comment]
Not sure what's wrong. Why there's no update for long time? Even no some automatic update form bots.

**Comment by bkjchoi72** [comment]
@guocb Working on this issue

**Comment by RonnyPfannschmidt** [comment]
it seems trace support hasn't been implemented in the unit-test plugin just like pdbinvoke had been added as extra step the pdbtrace support has to be added as well

**Comment by bkjchoi72** [comment]
@RonnyPfannschmidt I was about to open a PR for this, but thought it might be a good idea to run it by you. I was able to fix this issue by adding `self.ihook.pytest_pyfunc_call(pyfuncitem=self)` inside `unittest.py::TestCaseFunction::runtest` as following (https://github.com/pytest-dev/pytest/compare/master...bkjchoi72:3823-add-trace-option-for-unittest#diff-1d1f5a0728f7c0628e2fa4d542a9eca4R186). This allows unittest test cases to go into pdb, but I'm not sure if this will result in any side effects. As far as testing g

**Comment by RonnyPfannschmidt** [comment]
@bkjchoi72 we had a very critical reason to avoid that - large parts of the unittest machinery ely on correctly triggering its full behavior wrt setup/tear-down,

so from fuzzy memory i believe this breaks various aspects of unitest behavior for tracing

**Comment by blueyed** [comment]
@bkjchoi72 
Have you done more work in this regard already?

**Comment by blueyed** [comment]
Nevermind, found https://github.com/pytest-dev/pytest/pull/4083.

**Comment by eblume** [comment]
This issue still exists. Is the intention what this will not be resolved?

**Comment by The-Compiler** [comment]
> Based on the history though it sounds like it's still an issue.

Not sure what makes you think so - the fix which closed this issue (04f27d4eb469fb9c76fd2d100564a0f7d30028df) mentions fixing `--trace` with unittest and even includes a test for that.
